### PR TITLE
moved GetSkill function to more appropriate ItemTemplate class

### PR DESCRIPTION
**Pull request by Yehonal**
**Changes proposed**:

GetSkill is an Item function that just do checks on ItemTemplate, so it can be easily moved under ItemTemplate class and be used when you just have a prototype object

**Target branch(es)**: 335

**Tests performed**: Compilation works and supports legacy code

### CODE REVIEW

File: src/server/game/Entities/Item/ItemPrototype.h [hunk 1]
```diff
@@ -737,6 +737,41 @@ struct ItemTemplate
         return std::max<float>(0.f, itemLevel);
     }
 
+    uint32 GetSkill() const
```
Comment (jackpoz) [inline src/server/game/Entities/Item/ItemPrototype.h]: unrelated to this PR and related to the existing code too: what is the point of placing all the logic in the header ?

Comment (Yehonal) [inline src/server/game/Entities/Item/ItemPrototype.h]: Indeed, i've placed it there to avoid different ItemTemplate "code-standard"

**Comment by jackpoz** [comment]
This PR looks fine. The merger should move the logic to .cpp and cleanup the .h while at it.

It will probably have to wait for next TDB release before being merged tho.

**Comment by Naios** [comment]
Note that variables only placed in header files could lead to multiple instances of this variable when linking dynamically.

**Pull request closed without merging**